### ISSUE TITLE
/e in search is broken when using regular expressions

### ISSUE THREAD

**Issue by robertszalapski** [vrapper/vrapper#539]
For example, if I want to search for any numbers, and want to land on the end of that number:
/\s[0-9]+/e

The current behavior appears that /e is just doing /b+(searchExperssionLength-1).  If I search:
/ss/e
the cursor appears at the end, as expected.  But if I search
/[s]s/e
the cursor appears two characters later.

**Comment by albertdev** [comment]
I think I got it fixed now, but we haven't got tests for it yet. My manual tests appear to work fine though, even with `e+10` and so on.

The only remaining annoyance I've noticed is that if you `set selection=exclusive`, a visual selection will be off by one compared to Vim: Vim doesn't select the last character of the match, Vrapper does. It must be a corner case in Vim which we don't implement, but frankly I can't be bothered to implement this special case in Vrapper as it only complicates the logic for operator commands like `dn`.
